### PR TITLE
Add KeybindText

### PR DESCRIPTION
**Pull request by kashike**
**SpongeAPI** | [SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/1765)

### CODE REVIEW

File: src/main/java/org/spongepowered/api/text/KeybindText.java [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    KeybindText(String keybind) {
+        this.keybind = checkNotNull(keybind, "keybind");
+    }
+
+    /**
+     * Constructs a new immutable {@link LiteralText} for the keybind
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/text/KeybindText.java]: javadoc fail much?

File: src/main/java/org/spongepowered/api/text/KeybindText.java [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+import org.spongepowered.api.text.format.TextStyle;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a {@link Text} containing a keybind.
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/text/KeybindText.java:45]: a keybind that is documented by saying key bind makes a lot of sense. How do I use a key bind? "Oh, just make a key bind and use it!"

**Comment by stephan-gh** [comment]
Is an empty keybind text valid? Otherwise we should check this and eventually make the parameter required for the builder already (this is how it seems to be for the other text classes).

**Comment by Cybermaxke** [comment]
Maybe there should be a `CatalogType` for the supported keys? There is also a `key.invalid` key that can be used in cases that the key binding is unknown.

**Review by gabizou — CHANGES_REQUESTED** [review]
WHAT IS A KEYBIND TEXT?! Almost none of the descriptions so far have told me how it's used And the PR has no description, so that's a plus.

File: src/main/java/org/spongepowered/api/text/KeybindText.java (hunk 2 above)
Comment (ryantheleach) [reply src/main/java/org/spongepowered/api/text/KeybindText.java:45]: Represents a {`@link` Text} placeholder that on the client, replaces string keynames, eg: "key_key.swapHands" will be replaced with "F" on the client, see TranslatableText.

`/tellraw @a ["",{"text":"Press "},{"keybind":"key.swapHands"},{"text":" to pay respects"}]`
Press F to pay respects.

**Comment by Aaron1011** [comment]
What's the status of this PR? With the way that the Text implementation is currently setup, we can't properly handle mods using `TextComponentKeybind`, since we can't represent it as a `Text`.

**Comment by ryantheleach** [comment]
LGTM as-is (havn't checked it out and tested locally)

Is there something blocking this from being merged? @gabizou Your complaint earlier about the JD is just, but I don't think it should hold this up from merging.

**Comment by gabizou** [comment]
I think this will be API 8, if expected to be available then as Text has moved to implementation.

**Comment by ryantheleach** [comment]
@gabizou note, it's needed for stable as well, due to "What's the status of this PR? With the way that the Text implementation is currently setup, we can't properly handle mods using TextComponentKeybind, since we can't represent it as a Text."

It caused aaron various issues.

**Comment by Zidane** [comment]
@kashike Is this complete? Or do you plan to work on this anymore?

**Comment by Zidane** [comment]
Closing as we're going in a slightly different direction in api-8 and won't be adding major changes to api-7 at this point.

**Pull request closed without merging**